frame/support/procedural/src/pallet/expand/storage.rs

```diff
@@ -122,27 +138,81 @@ pub fn process_generics(def: &mut Def) -> syn::Result<()> {
 
 		let default_query_kind: syn::Type =
 			syn::parse_quote!(#frame_support::storage::types::OptionQuery);
-		let default_on_empty: syn::Type = syn::parse_quote!(#frame_support::traits::GetDefault);
+		let mut default_on_empty = |value_ty: syn::Type| -> syn::Type {
+			if let Some(QueryKind::ResultQuery(variant_name)) = storage_def.query_kind.as_ref() {
+				let on_empty_ident = quote::format_ident!("Get{}Result", storage_def.ident);
```

Comment:
```suggestion
				let on_empty_ident = quote::format_ident!("__Frame_Internal_Get{}Result", storage_def.ident);
```

Or whatever, but something more "unique"